app/models/appointment.js

```diff
@@ -87,7 +87,36 @@ export default AbstractModel.extend({
       presence: true
     },
     startDate: {
-      presence: true
+      acceptance: {
+        accept: true,
+        if(object) {
+          if (!object.get('hasDirtyAttributes')) {
+            return false;
+          }
+          let allDay = object.get('allDay');
+          let startDate = object.get('startDate');
+          let endDate = object.get('endDate');
+          if (isEmpty(startDate)) {
+            // force validation to fail
+            return true;
+          } else if  (!isEmpty(endDate)) {
+            // end date is already selected; do further validation
+            if (allDay) {
+              if (endDate.getTime() < startDate.getTime()) {
+                return true;
+              }
+            } else {
+              if (endDate.getTime() <= startDate.getTime()) {
+                return true;
+              }
+            }
+          }
+          // patient is properly selected; don't do any further validation
+          return false;
+
+        },
+        message: 'Please select a start date earlier than the end date'
```

Comment:
Can we use i18n here?

Reply:
The other messages in that file don't seem to use it. Should I change all of them?

Reply:
Absolutely! I'll do both.

Reply:
You can probably do the function the same way as the visit.js model has it with the dateAcceptance function.